src/core/cgroup.c

```diff
@@ -3571,12 +3571,17 @@ void unit_prune_cgroup(Unit *u) {
         if (!crt || !crt->cgroup_path)
                 return;
 
-        /* Cache the last CPU and memory usage values before we destroy the cgroup */
+        /* Cache the last resource usage values before we destroy the cgroup */
         (void) unit_get_cpu_usage(u, /* ret = */ NULL);
 
         for (CGroupMemoryAccountingMetric metric = 0; metric <= _CGROUP_MEMORY_ACCOUNTING_METRIC_CACHED_LAST; metric++)
                 (void) unit_get_memory_accounting(u, metric, /* ret = */ NULL);
 
+        /* All IO metrics are read at once from the underlying cgroup, so issue just a single call */
+        (void) unit_get_io_accounting(u, /* metric = */ 0, /* allow_cache = */ false, /* ret = */ NULL);
```

Comment:
Ugh, I find the `allow_cache` name rather confusing BTW, as if the cgroup has been destroyed we'd read from cache regardless. Could you rename it to `prefer_cache` or `shortcut_cache`? Otherwise I'll submit a PR later.

Reply:
Ah, nevermind, I forgot that it will still go through the cgroup if the cache is invalid.

Reply:
Actually, given we're bikeshedding the name... I think I don't like `prefer_cache` after all, because it does not properly reflect the fact that we can and will get stale values. Other ideas? :sweat_smile:

Reply:
Hmm, looking at it a bit closer, I think it's OK to just remove it? `unit_get_cpu_usage` and `_memory_accounting` both don't carry such a param. Effectively this only saves one `unit_get_io_accounting_raw` call which immediately returns `-ENODATA` here anyway.

Reply:
Yep, you're right. The regular DBus unit property path does not make use of `allow_cache` anyway, which means the complexity is already `O(n^2)` most of the time.